x/cork/keeper/query_server.go

```diff
@@ -92,3 +62,31 @@ func (k Keeper) QueryScheduledCorksByBlockHeight(c context.Context, req *types.Q
 	response.Corks = k.GetScheduledCorksByBlockHeight(ctx, req.BlockHeight)
 	return &response, nil
 }
+
+func (k Keeper) QueryScheduledCorksByID(c context.Context, req *types.QueryScheduledCorksByIDRequest) (*types.QueryScheduledCorksByIDResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	response := types.QueryScheduledCorksByIDResponse{}
+	response.Corks = k.GetScheduledCorksByID(ctx, req.Id)
+	return &response, nil
+}
```

Comment:
So if I'm understanding right, this can return several sets of scheduled corks. Let's say N SPs have submitted a unique cork around the same time, so there are N sets of scheduled corks all of id 0. From a user perspective what's the value of querying id `0` and getting back more than one "scope"?

Reply:
The ID is a hash of the scope and the block_height. This means that corks submitted at the same height, in the same scope, will map to the same cork ID. Therefore, it's impossible to query an ID and get corks of more than one scope.

Reply:
The ID is just an `int`, and it always starts at 0 for a new scope

Reply:
I see where I was mistaken, I think I get it now thanks!